src/js/views/patients/schedule/schedule_views.js

```diff
@@ -364,6 +372,55 @@ const ScheduleListView = CollectionView.extend({
     }, []);
     this.triggerMethod('filtered', visibleActions);
   },
+  onSelect(selectedView, isShiftKeyPressed, selectedDayIndex) {
+    const isSelected = this.state.isSelected(selectedView.model);
+
+    const sortedArrayOfActions = this.getSortedArrayOfActions();
+    const indexOfSelectedAction = sortedArrayOfActions.map(action => action.id).indexOf(selectedView.model.id);
+
+    const lastSelectedIndex = this.state.get('lastSelectedIndex');
+
+    if (isShiftKeyPressed && lastSelectedIndex !== null && !isSelected) {
+      this.handleClickShiftMultiSelect(sortedArrayOfActions, indexOfSelectedAction, lastSelectedIndex);
+      return;
+    }
+
+    this.state.toggleSelected(selectedView.model, !isSelected, indexOfSelectedAction);
+  },
+  handleClickShiftMultiSelect(sortedArrayOfActions, indexOfSelectedAction, lastSelectedIndex) {
+    const minIndex = Math.min(indexOfSelectedAction, lastSelectedIndex);
+    const maxIndex = Math.max(indexOfSelectedAction, lastSelectedIndex);
+
+    const selectedIds = sortedArrayOfActions.map(action => action.id).slice(minIndex, maxIndex + 1);
+
+    this.state.selectMultiple(selectedIds, indexOfSelectedAction);
+  },
+  getSortedArrayOfActions() {
+    const collectionJSON = this.collection.toJSON();
+
+    const sortedCollectionOfDays = collectionJSON.sort(function(a, b) {
+      return new Date(a.date) - new Date(b.date);
+    });
+
+    const sortedArrayOfActions = [];
+
+    each(sortedCollectionOfDays, day => {
+      const actions = day.actions.toJSON();
+
+      const sortedCollectionOfActions = actions.sort(function(a, b) {
+        if (!a.due_time) a.due_time = '24';
+        if (!b.due_time) b.due_time = '24';
+
+        return a.due_time.localeCompare(b.due_time);
+      });
+
+      each(sortedCollectionOfActions, action => {
+        sortedArrayOfActions.push(action);
+      });
+    });
+
+    return sortedArrayOfActions;
+  },
```

Comment:
To get the correct index position of a selected list item, I needed to create a flattened version of the collection and sort those actions by date.

Convert this:

```js
[
  {
    day: {
      actions: [ {action}, {action} ]
    }
  },
  {
    day: {
      actions: [ {action} ]
     }
   }
]
```

Into this:

```js
[
  {action}, {action}, {action}
]
```

Where actions are sorted by due day and time.